README.md

```diff
@@ -158,9 +166,599 @@ search.start();
+
+Require InstantSearch v4.8.3 or later
+
+- ✅ insightsClient: Insight Client uses [search-insight.js](https://github.com/algolia/search-insights.js).
+- ✅ insightsInitParams: Insight params.
+- ✅ onEvent?: function triggered on events.
+
+### ✅ Middleware
+
+[Middleware references](https://www.algolia.com/doc/api-reference/widgets/middleware/js/)
+
+Middleware is a function returning an object with `onStateChange`, `subscribe` and `unsubscribe` functions. With the middleware API, you can inject functionalities in the InstantSearch lifecycle.
+
+### ✅ renderState
+
+[renderState references](https://www.algolia.com/doc/api-reference/widgets/render-state/js/#examples)
+
+Provides all the data and functions from the widgets.
+
+It works only on widgets that are compatible with instantMeiliSearch.
+
+### ✅ Hits
+
+[hits references](https://www.algolia.com/doc/api-reference/widgets/hits/js/)
+
+Used to display a list of results.
+
+- ✅ container: The CSS Selector or HTMLElement to insert the hits into. _required_
+- ✅ escapeHTML: Escapes HTML tags in string values in returned hits.
+- ✅ transformItems: Function that maps over every hit the provided logic.
+- ✅ templates: The templates to use for the widget.
+- ✅ cssClasses: The CSS classes to override.
+
+```js
+instantsearch.widgets.hits({
+  container: '#hits',
+  templates: {
+    item: `
+        <div>
+          <div class="hit-name">
+            {{#helpers.highlight}}{ "attribute": "title" }{{/helpers.highlight}}
+          </div>
+        </div>
+      `,
+  },
+})
+```
+
+### ✅ InfiniteHits
+
+[infiniteHits references](https://www.algolia.com/doc/api-reference/widgets/infinite-hits/js/)
+
+The `infiniteHits` widget is used to display a list of results with a “Show more” button.
+
+- ✅ container: The CSS Selector or HTMLElement to insert the hits into. _required_
+- ✅ escapeHTML: Escapes HTML tags in string values in returned hits.
+- ✅ transformItems: Function that maps over every hit the provided logic.
+- ✅ templates: The templates to use for the widget.
+- ✅ cssClasses: The CSS classes to override.
+- ❌ showPrevious: Did not work with both Algoliasearch and InstantMeiliSearch.
+- ❌ cache: Not added in InstantMeiliSearch.
+
+```js
+instantsearch.widgets.infiniteHits({
+  container: '#infinite-hits',
+  templates: {
+    item: `
+      <h2>
+        {{#helpers.highlight}}{ "attribute": "name" }{{/helpers.highlight}}
+      </h2>
+    `,
+  },
+})
+```
+
+### ✅ Highlight
+
+[highlight references](https://www.algolia.com/doc/api-reference/widgets/highlight/js/)
+
+The `highlight` function returns an attribute from a hit into its highlighted form, when relevant.
+
+- ✅ attribute: The attribute of the record to highlight. _required_
+- ✅ hit: Hit object. _required_
+- ✅ highlightedTagName: HTML element to wrap the highlighted parts of the string.
+
+See [Hits](#hits) for an example.
+
+### ✅ Snippet
```

Comment:
Can you precise it corresponds to the "Cropping"  of MeiliSearch?

Reply:
And the crop in MeiliSearch is in char, not in word like Algolia does.
`["overview:20"]` is not the same as with Algolia.

Reply:
It works the same way with algolia I think: https://codesandbox.io/s/jolly-hypatia-xtkv1?file=/src/App.js
Could you confirm with this codesandbox ?

Reply:
- change `description:20` by `description:2` -> you will get 2 words, and only 1 with MeiliSearch
- https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/highlighting-snippeting/js/#nbwords

Reply:
Ah I thought you talked about usage and not how features works with algolia's compared to meilisearch.

The difference happens in algolia's core. See [this issue]((https://github.com/algolia/react-instantsearch/issues/3046#issuecomment-827719846)) that I think confirms that this happens in their core. and is thus may not be related to compatibility with`instantSearch.js` components. 

Should we still document this here or make a link to the issue that details the differences between `meilisearch` and `algoliasearch` (see #389)?

Maybe we could say something like 
```
Note that `snippeting` works differently with the `algoliasearch-client-javascript` client than with `instant-meilisearch` client. Precise comparison is [given in this issue](https://github.com/meilisearch/instant-meilisearch/issues/389).
```